perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -152,8 +221,10 @@ def Cleanup(benchmark_spec):
     benchmark_spec: The benchmark specification. Contains all data that is
         required to run the benchmark.
   """
-  vms = benchmark_spec.vms
-  vm = vms[0]
+  vm = benchmark_spec.vms[0]
   logging.info('FIO Cleanup up on %s', vm)
-  vm.RemoveFile(flags.FLAGS.fio_jobfile)
-  vm.RemoveFile(vm.GetScratchDir() + flags.FLAGS.fio_benchmark_filename)
+  vm.RemoveFile(REMOTE_JOB_FILE_PATH)
+  if not FLAGS.against_device and not FLAGS.fio_jobfile:
+    # If the user supplies their own job file, then they have to clean
+    # up after themselves, because we don't know their temp file name.
+    vm.RemoveFile(vm.GetScratchDir() + 'fio-temp-file')
```

Comment:
We don't know their temp file name because we would have to parse their job file in order to know it. It would add a lot of complexity to handle all of the cases correctly, so I think it's best to just say that if the user supplies a file, they can handle cleanup. It shouldn't be too big a deal, because in the common case we just destroy the scratch disk after testing and the file goes away.